pkg/activator/handler/handler_test.go

```diff
@@ -266,7 +266,7 @@ func TestActivationHandlerTraceSpans(t *testing.T) {
 
 func sendRequest(namespace, revName string, handler http.Handler, store *activatorconfig.Store) *httptest.ResponseRecorder {
 	resp := httptest.NewRecorder()
-	req := httptest.NewRequest(http.MethodPost, "http://example.com", nil)
+	req := httptest.NewRequest(http.MethodPost, "http://example.com/", nil)
```

Comment:
Is this change necessary? If so, does this break anything?

Reply:
yeah it's a bit silly but basically the URLs are the same (the canonical version of the URL contains the slash) and it's impossible to actually _send_ a request with an empty path (actually sending the request on the wire would add it), but if you create it manually and directly call ServeHTTP like this you can get a req.URL with no path passed to ServeHTTP, and if so [this test](https://github.com/knative/serving/blob/f8325b7541362dd0550e8d69e9b539b88b0ea114/pkg/activator/handler/handler_test.go#L196) will fail because the span looks for the path to be "/" on [L204](https://github.com/knative/serving/blob/f8325b7541362dd0550e8d69e9b539b88b0ea114/pkg/activator/handler/handler_test.go#L204). NewSingleHostReverseProxy happens to add a '/' in this case when it [munges the target.URL.Path and req.URL.Path together](https://github.com/golang/go/blob/9c017ff30dd21bbdcdb11f39458d3944db530d7e/src/net/http/httputil/reverseproxy.go#L130), so it was re-introducing the '/'. Could have added a line to NewHeaderPruningReverseProxy to prefix empty paths but it seems better to fix the test since it can't happen in practice.

Reply:
Does HTTP actively forbid calls without a path?

Reply:
yeah, ["Note that the absolute path cannot be empty; if none is present in the original URI, it MUST be given as "/" (the server root)."](https://www.w3.org/Protocols/rfc2616/rfc2616-sec5.html)

Reply:
Cool, alrighty then!

Reply:
(also see: https://play.golang.org/p/OrdIOT3jQaj)